### PR TITLE
chore: Make the qns image multiarch & shrink the build context further

### PR DESCRIPTION
**Pull request by larseggert**
Also use `git` to get nss, which is a bit faster.

CC @mxinden

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,3 +1,8 @@
-nss
-nspr
-target
+# Ignore everything:
+*
+# Except for the following:
+!**/*.toml
```
Comment (mxinden) [inline .dockerignore:4]:
```suggestion
!**/*.toml
!Cargo.lock
```

Easier for debugging, as both a local `cargo xxx` as well as the `cargo build`  in the `Dockerfile` will use the same dependencies.

**Review by mxinden — APPROVED** [review]
:+1:

File: qns/Dockerfile [hunk 2]
```diff
@@ -14,19 +14,15 @@ ENV RUSTUP_HOME=/usr/local/rustup \
     PATH=/usr/local/cargo/bin:$PATH
 
 RUN set -eux; \
```
Comment (martinthomson) [inline qns/Dockerfile:16]: I'm less of a fan of dropping the checksum, but I can see the value.

```suggestion
RUN \
```

You don't need the set call (unless you want to call `set -o pipefail`).

**Review by martinthomson — DISMISSED** [review]

File: qns/Dockerfile [hunk 3]
```diff
@@ -13,20 +13,16 @@ ENV RUSTUP_HOME=/usr/local/rustup \
...
-RUN set -eux; \
-    curl -sSLf "https://static.rust-lang.org/rustup/archive/1.26.0/x86_64-unknown-linux-gnu/rustup-init" -o rustup-init; \
-    echo '0b2f6c8f85a3d02fde2efc0ced4657869d73fccfce59defb4e8d29233116e6db *rustup-init' | sha256sum -c -; \
-    chmod +x rustup-init; \
-    ./rustup-init -y -q --no-modify-path --profile minimal --default-toolchain "$RUST_VERSION"; \
-    rm -f rustup-init; \
-    chmod -R a+w "$RUSTUP_HOME" "$CARGO_HOME"
+RUN \
+    curl https://sh.rustup.rs -sSf | \
+    sh -s -- -y -q --no-modify-path --profile minimal --default-toolchain $RUST_VERSION
```
Comment (larseggert) [inline qns/Dockerfile:18]:
```suggestion
RUN curl https://sh.rustup.rs -sSf | \
    sh -s -- -y -q --no-modify-path --profile minimal --default-toolchain $RUST_VERSION
```

File: .dockerignore [hunk 4]
```diff
@@ -1,3 +1,9 @@
...
+!Cargo.lock
+!**/*.rs
+!**/*.h
+!**/*.hpp
+!qns
```
Comment (larseggert) [inline .dockerignore:9]:
```suggestion
!**/*.toml
!**/*.rs
!**/*.h
!**/*.hpp
!qns
!Cargo.lock
```

**Pull request merged**